text/0141-linking-traces.md

```diff
@@ -0,0 +1,361 @@
+- Start Date: 2024-11-08
+- RFC Type: feature 
+- RFC PR: <[link](https://github.com/getsentry/rfcs/pull/141)>
+- RFC Status: draft
+
+# Summary
+
+This RFC introduces the ability to link or connect traces in Sentry. The required changes involve SDKs, Ingest, Search and Storage, Backend as well as product teams.
+The goal of this initiative is to be able to link multiple traces together so that the Sentry product can show what happened before a specific trace while preserving the integrity of individual traces.
+This RFC proposes to use "Span Links" as a vehicle for specifying relationships between spans in multiple traces. It also discusses other options considered, as well as concrete use cases for this feature.
+
+# Motivation
+
+At the time of writing this RFC, Sentry (the product as well as SDKs) cannot connect multiple traces to deliver a "bigger picture" of what happened before (or after) a specific trace. 
+A trace is therefore the highest level of connecting signals (errors, traces, profiles, etc.).
+The notable exception is Session Replay, where we are indeed able to provide this bigger picture.
+
+This RFC proposes a method of achieving linkage between (and within) traces in the entire Sentry product, to satisfy a number of currently sub-optimally solved use cases.
+
+## Frontend Traces and User Journeys
+
+The most important application of linking traces are frontend applications are frontend applications. We would like to display a user journey (session) which should make debugging of issues much easier as developers have more context on what happened before a specific issue. 
+Today, we are limited to the duration of one trace (id), which is handled differently and kept alive for different times across our various SDKs. A concrete example are our [JavaScript Browser SDKs](https://develop.sentry.dev/sdk/platform-specifics/javascript-sdks/browser-tracing/#tracing-model) which by default keep a trace alive as long as users
+are on the same page our (URL) route. This was a compromise in which we accepted that a trace would consist of multiple trace root spans (transactions), which is generally discouraged by tracing models like OpenTelemetry.
+Another example of a suboptimal trace model is the one used in most mobile SDKs. In these SDKs we 
```

Comment:
something is missing here

Reply:
thx, went with https://github.com/getsentry/rfcs/pull/141#discussion_r1844089805